### PR TITLE
[AIP-294] Update intake_online dal to read Json

### PR DESCRIPTION
**Pull request by smatzana**
Adapt intake-dal to read the new Json based format the fs-web-api is now returning

### CODE REVIEW

File: intake_dal/tests/test_dal_online.py [hunk 1]
```diff
@@ -27,33 +26,52 @@ def user_events_df():
...
 
     mock_get.assert_called()
     mock_put.assert_called()
 
-    assert_frame_equal(
-        user_events_df,
-        deserialize_avro_str_to_pandas(mock_put.call_args[0][1]["avro_rows"]),
-        check_dtype=False,
-    )
-
```
Comment (talebzeghmi) [inline intake_dal/tests/test_dal_online.py]: The check on `mock_put.call_args[0][1]["avro_rows"]` ensures that intake-dal write is sending the correct avro to mock_put-> _http_put_avro_data_set.  Removing that line would remove that validation.  Ideally I should have made that a separate unit test.

File: intake_dal/tests/test_dal_online.py [hunk 2]
```diff
@@ -27,33 +26,52 @@ def user_events_df():
...
     )
-    mock_get.return_value = avro_str
+    mock_get.return_value = user_events_json
     mock_put.return_value = 200
 
     serving_cat.entity.user.user_events(storage_mode="serving").write(user_events_df)
     assert serving_cat.entity.user.user_events(key=100).read().iloc[0].userid == 100
     assert serving_cat.entity.user.user_events(key=100).read().iloc[0].home_id == 3
 
-    assert_frame_equal(user_events_df, serving_cat.entity.user.user_events(key=100).read(), check_dtype=False)
```
Comment (talebzeghmi) [inline intake_dal/tests/test_dal_online.py]: I could have removed line 42 and 44 here and use only `assert_frame_equal()`. I think that's cleaner.

File: intake_dal/dal_online.py [hunk 3]
```diff
@@ -72,9 +73,12 @@ def get_metadata(key: str, default):
 
     def _get_partition(self, _) -> pd.DataFrame:
         self._get_schema()
-        return deserialize_avro_str_to_pandas(
-            _http_get_avro_data_set(self._url, self._canonical_name, self._key_value), self._avro_schema
-        )
+        data = _http_get_avro_data_set(self._url, self._canonical_name, self._key_value)
+        for item in data:
```
Comment (talebzeghmi) [inline intake_dal/dal_online.py]: item is a row.

File: intake_dal/dal_online.py [hunk 4]
```diff
@@ -72,9 +73,12 @@ def get_metadata(key: str, default):
...
+            for key, field in item.items():
+                if isinstance(field, dict) and "DATETIME" in field:
+                    item[key] = datetime.strptime(field["time"])
+        return pd.DataFrame(data)
```
Comment (talebzeghmi) [inline intake_dal/dal_online.py]: A nice to have is a unit test that measures the performance of JSON -> pandas to ensure it doesn't regress, or at the least print out the performance.

File: intake_dal/tests/test_dal_online.py (hunk 1 above)
Comment (smatzana) [reply intake_dal/tests/test_dal_online.py]: added check back in

File: intake_dal/tests/test_dal_online.py (hunk 2 above)
Comment (smatzana) [reply intake_dal/tests/test_dal_online.py]: agree! Fixed

File: intake_dal/dal_online.py (hunk 3 above)
Comment (smatzana) [reply intake_dal/dal_online.py]: fixed

**Comment by smatzana** [comment]
Most comments addressed (except the nice to have :) )

File: intake_dal/dal_online.py [hunk 5]
```diff
@@ -25,6 +26,8 @@ class DalOnlineSource(DataSource):
     name = "dal-online"
     version = pkg_resources.get_distribution("intake-dal").version
 
+    __date_time_format = "%Y-%m-%d %H:%M:%S.%f"
```
Comment (talebzeghmi) [inline intake_dal/dal_online.py]: Let's make this a constant with uppercase like `GLOBAL_CONSTANT_NAME` in http://google.github.io/styleguide/pyguide.html

Comment (smatzana) [reply intake_dal/dal_online.py]: done

**Pull request merged**